app/screens/home/search/search.tsx

```diff
@@ -118,15 +130,21 @@ const SearchScreen = ({teamId}: Props) => {
         };
     }, [headerHeight, lastSearchedValue]);
 
+    const onClear = useCallback(() => {
+        setSearchValue('');
+        setLastSearchedValue('');
+        setFilter(FileFilters.ALL);
+    }, [filter]);
+
     let header = null;
     if (lastSearchedValue) {
         header = (
             <Header
                 onTabSelect={setSelectedTab}
-                onFilterChanged={setFilter}
+                onFilterChanged={handleFilterChange}
                 numberMessages={postIds.length}
                 selectedTab={selectedTab}
-                numberFiles={Object.keys(filteredFileInfos).length}
+                numberFiles={Object.keys(fileInfos).length}
```

Comment:
fileInfos is already an array, not sure why you are using `Object.keys`

```suggestion
                numberFiles={fileInfos.length}
```